referenceframe/frame_system.go

```diff
@@ -308,6 +314,74 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD takes in a framesystem and returns a map where all elements are
+// the point representation of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input, logger *zap.SugaredLogger) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	geoMap, err := FrameSystemGeometries(system, inputs, logger)
+	if err != nil {
+		return nil, err
+	}
+	for name, geosInFrame := range geoMap {
+		geos := geosInFrame.geometries
+		aggregatePoints := []r3.Vector{}
+		for _, g := range geos {
+			asPoints := g.ToPoints(defaultPointDensity)
+			aggregatePoints = append(aggregatePoints, asPoints...)
+		}
+		vectorMap[name] = aggregatePoints
+	}
+	return vectorMap, nil
+}
+
+// FrameSystemGeometries takes in a framesystem and returns a map where all elements
+// are GeometriesInFrame modified to be with respect to the world.
+//
+//nolint:lll
+func FrameSystemGeometries(system FrameSystem, inputs map[string][]Input, logger *zap.SugaredLogger) (map[string]*GeometriesInFrame, error) {
+	geoMap := make(map[string]*GeometriesInFrame)
+	for _, name := range system.FrameNames() {
+		currentFrame := system.Frame(name)
+		currentInput := inputs[name]
+		// if currentInput is nil and DoF != 0 we chose to omit the
+		// frame entirely as this would return the frame's geometries
+		// in their home or "zero" position, and not in their
+		// current position.
+		if currentInput == nil && len(currentFrame.DoF()) == 0 {
+			currentInput = []Input{}
+		}
+		if currentInput == nil {
+			logger.Debugf("will not transform %v to be with respect to the world", name)
+			continue
+		}
+		geosInFrame, err := currentFrame.Geometries(currentInput)
+		if err != nil {
+			return nil, err
+		}
+		if len(geosInFrame.Geometries()) > 0 {
+			// the parent of the frame is handled by the Transform method.
+			transformed, err := system.Transform(inputs, geosInFrame, World)
+			if err != nil && strings.Contains(err.Error(), "no positions provided for frame with name") {
+				logger.Debugf("unable to handle the transform for %v", name)
```

Comment:
The various debug statements below here are complex and also unneeded- it may not be the direct parent that had no inputs, rather the grandparent, or great-grandparent, etc.

Something that will capture the needed information and also will be much more succinct  would be to append the text of `err.Error()` to this debug statement.